tests/integration_tests/test_system_stats.py

```diff
@@ -25,38 +25,37 @@ def teardown_module():
 class TestSystemStats:
     def test_system_stats(self):
         ws = hub.connect()
-        try:
-            hub.send_identity(ws, "system stats test")
-            hub.send_subscribe(ws, ["system_stats"])
-
-            response = hub.recv(ws)
-            assert response["type"] == "iseeu"
-
-            # after the first iseeu message all messages should be state updates
-
-            last_host_name = None
-            for _i in range(5):
-                message = hub.recv(ws)
-                assert message["type"] == "stateUpdate"
-
-                cpu_util = message["data"]["system_stats"]["cpu_util"]
-                assert cpu_util >= 0
-                assert cpu_util <= 100
-
-                # Doing this test on the CI/CD runner is not reliable
-                #   because the running environment is reporting the
-                #   CPU utilization of the runner itself, not the service.
-                # assert cpu_util != last_cpu_util
-                # last_cpu_util = cpu_util
-
-                host_name = message["data"]["system_stats"]["hostname"]
-                assert host_name
-                if last_host_name:
-                    assert host_name == last_host_name, "host name should not change"
-                last_host_name = host_name
-
-                # as long as the service sleeps the same amount of time that
-                # the test sleeps we should get a new system stats message each time
-                time.sleep(TEST_SAMPLE_INTERVAL)
-        finally:
-            ws.close()
```

Comment:
this was causing the test to hang on my mac.  pytest was just hanging after reporting all success on integration tests and I'm pretty sure it was because the hub_state_monitor thread was still running.  All of the other tests were doing the `ws.close()`, but NOT doing it in a `try ... finally`